### PR TITLE
Bug 1885856: Exporting registry v1 protocol usage metric

### PR DESCRIPTION
**Pull request by ricardomaraschini**
This PR exports v1_image_imports_total metric. We use this metric to
count the number of image streams imported using registry v1 as we are
planning to deprecate this version soon.

* [ ] I added CHANGELOG entry for this change.
* [x] No user facing changes, so no entry in CHANGELOG was needed.

### CODE REVIEW

File: Documentation/data-collection.md [hunk 1]
```diff
@@ -236,6 +236,10 @@ data:
     - '{__name__="che_workspace_start_time_seconds_sum"}'
     # (codeready workspaces, @ibuziuk) The overall number of attempts for starting all the workspaces.
     - '{__name__="che_workspace_start_time_seconds_count"}'
+    # (image-registry-team) v1_image_imports_total counts the number of image stream imports made
+    # using registry protocol v1, we are moving forward with deprecating this version hence we
+    # need to know how often it is used.
+    - '{__name__="v1_image_imports_total"}'
```
Comment (lilic) [inline Documentation/data-collection.md:242]: Is not the full name of the metric `apiserver_v1_image_imports_total`?

**Comment by lilic** [comment]
Note in case you missed it here are the full docs around how to get approval on this https://docs.google.com/document/d/1a6n5iBGM2QaIQRg9Lw4-Npj6QY9--Hpx3XYut-BrUSY/edit

tl;dr: You need to get Claytons 👍 on it, we already had a look at the actual metric. Let me know if any step don't make sense, happy to clarify! :)

File: Documentation/data-collection.md (hunk 1 above)
Comment (ricardomaraschini) [reply Documentation/data-collection.md:242]: Thanks @lilic! I wasn't sure if I should or not add the "subsystem" into it, going to fix it right now.

Comment (lilic) [reply Documentation/data-collection.md:242]: Name here should be as it appears in Prometheus, as this is just a match selector for the metrics (series) themselves.

**Comment by ricardomaraschini** [comment]
> Note in case you missed it here are the full docs around how to get approval on this https://docs.google.com/document/d/1a6n5iBGM2QaIQRg9Lw4-Npj6QY9--Hpx3XYut-BrUSY/edit
> 
> tl;dr: You need to get Claytons  on it, we already had a look at the actual metric. Let me know if any step don't make sense, happy to clarify! :)

Thanks for helping me on this one!

File: Documentation/data-collection.md [hunk 2]
```diff
@@ -236,6 +236,10 @@ data:
     - '{__name__="che_workspace_start_time_seconds_sum"}'
     # (codeready workspaces, @ibuziuk) The overall number of attempts for starting all the workspaces.
     - '{__name__="che_workspace_start_time_seconds_count"}'
+    # (image-registry-team) apiserver_v1_image_imports_total counts the number of image stream
+    # imports made using registry protocol v1, we are moving forward with deprecating this version
+    # hence we need to know how often it is used.
+    - '{__name__="apiserver_v1_image_imports_total"}'
```
Comment (lilic) [inline Documentation/data-collection.md:242]: Is this by default in every openshift, as I did not see it on my cluster?

Comment (ricardomaraschini) [reply Documentation/data-collection.md:242]: Yes it is by default in all clusters. It will show up as soon as you import an image from a image registry using protocol v1. This protocol version has already been deprecated by docker and we don't expect this metric to show up.

Comment (dmage) [reply Documentation/data-collection.md:242]: @lilic will it expose the metric labels? We need only sum(apiserver_v1_image_imports_total), labels may have sensitive data (names of private registries and repositories).

Comment (simonpasquier) [reply Documentation/data-collection.md:242]: You have to create a recording rule doing the sum() then.

Comment (ricardomaraschini) [reply Documentation/data-collection.md]: @simonpasquier I have sent over a new commit that adds `:sum` at the end of the metric we are querying. Could you please confirm this is the right way of achieving the desired outcome?

Comment (simonpasquier) [reply Documentation/data-collection.md]: Have you defined the recording rule that generates the `apiserver_v1_image_imports:sum` somewhere (typically in the operator managing the component emitting the `apiserver_v1_image_imports_total` metric)?

Comment (ricardomaraschini) [reply Documentation/data-collection.md]: @simonpasquier no, could you please point me to a sample where this is done?

**Comment by ricardomaraschini** [comment]
/hold 

we need a recording rule

File: Documentation/data-collection.md (hunk 2 above)
Comment (ricardomaraschini) [reply Documentation/data-collection.md]: @simonpasquier I have created the rule in here, could you take a look? https://github.com/openshift/cluster-image-registry-operator/pull/626

**Comment by lilic** [comment]
You still need Clayton's approval on it as per the doc, but this looks good to me!

**Comment by ricardomaraschini** [comment]
/retest

502 Bad Gateway

File: manifests/0000_50_cluster-monitoring-operator_04-config.yaml [hunk 3]
```diff
@@ -230,6 +230,10 @@ data:
     - '{__name__="che_workspace_start_time_seconds_count"}'
     # (cloud credential operator, @openshift/openshift-team-hive) Track current mode the cloud-credentials-operator is functioning under.
     - '{__name__="cco_credentials_mode"}'
+    # (image-registry-team) :apiserver_v1_image_imports:sum counts the number of image stream
+    # imports made using registry protocol v1, we are moving forward with deprecating this version
```
Comment (smarterclayton) [inline manifests/0000_50_cluster-monitoring-operator_04-config.yaml:234]: I would rephrase as

"we are deprecating this version of the protocol and will use this metric to identify how many users are impacted by deprecation and when usage has reached zero".

That way someone reading this (someone who cares what data we send) understands how we will use the data.

**Comment by wzheng1** [comment]
LGTM based on https://bugzilla.redhat.com/show_bug.cgi?id=1885856#c3

File: manifests/0000_50_cluster-monitoring-operator_04-config.yaml (hunk 3 above)
Comment (ricardomaraschini) [reply manifests/0000_50_cluster-monitoring-operator_04-config.yaml]: Changed, tks!

Comment (ricardomaraschini) [reply manifests/0000_50_cluster-monitoring-operator_04-config.yaml]: Can we have this merged?

**Comment by ricardomaraschini** [comment]
Could someone please LGTM this?

**Comment by lilic** [comment]
@ricardomaraschini you need a explicit approval from Clayton, after that we can do an lgtm. I don't see it from his comment https://github.com/openshift/cluster-monitoring-operator/pull/949#discussion_r510951511

File: Documentation/data-collection.md [hunk 4]
```diff
@@ -238,6 +238,10 @@ data:
     - '{__name__="che_workspace_start_time_seconds_count"}'
     # (cloud credential operator, @openshift/openshift-team-hive) Track current mode the cloud-credentials-operator is functioning under.
     - '{__name__="cco_credentials_mode"}'
+    # (image-registry-team) :apiserver_v1_image_imports:sum counts the number of image stream
```
Comment (smarterclayton) [inline Documentation/data-collection.md:241]: Why is there no prefix on this metric?  Was there a reason for that?

Comment (smarterclayton) [reply Documentation/data-collection.md:241]: I can't actually understand why the metric name is bad (i saw lili's question above)

Comment (dmage) [reply Documentation/data-collection.md:241]: @smarterclayton the name was suggested by @simonpasquier at https://github.com/openshift/cluster-image-registry-operator/pull/626#discussion_r503920668

Comment (smarterclayton) [reply Documentation/data-collection.md:241]: It looks like this is following a convention from upstream docs that we don't follow elsewhere, so we have one metric that is different from all the others.
